lua/telescope/_extensions/bibtex.lua

```diff
@@ -96,7 +118,25 @@ end
 
 return telescope.register_extension {
   setup = function(ext_config)
-    depth = ext_config.depth
+    depth = ext_config.depth or depth
+    local custom_formats = ext_config.custom_formats or {}
+    for _, format in pairs(custom_formats) do
+      formats[format.id] = format.cite_marker
+    end
+    user_format = ext_config.format or fallback_format
+    if formats[user_format] == nil then
+      user_format = fallback_format
+    end
+    local global_files = ext_config.global_files or {}
+    for _,file in pairs(global_files) do
+      local p = path:new(file)
+      if p:is_dir() then
+	getBibFiles(file)
+      elseif p:is_file() then
+	table.insert(files, {name = file, mtime = 0, entries = {} })
+      end
+    end
+    getBibFiles('.')
```

Comment:
This code will run on startup. when `load_extension('bibtex')` happens. That might be bad for performance if cwd is in a big project (also happens if you are currently working on a not bibtex related project) and depth is != 1/2. You have two options move that first caching at the beginning of the first time bibtex gets called or maybe `scan.scan_dir_async` might help, but i am not so sure about that.

Reply:
You're right this is suboptimal for startup time. Doing it the first time bibtex gets called, seems the better solution.